genie-core/src/main/java/com/netflix/genie/core/jpa/services/JpaApplicationServiceImpl.java

```diff
@@ -96,19 +113,19 @@ public JpaApplicationServiceImpl(
     @Override
     public String createApplication(
         @NotNull(message = "No application entered to create.")
-        @Valid
-        final Application app
+        @Valid final Application app
     ) throws GenieException {
         log.debug("Called with application: {}", app);
         final Optional<String> appId = app.getId();
-        if (appId.isPresent() && this.applicationRepo.exists(appId.get())) {
+        if (appId.isPresent() && this.applicationRepository.existsByUniqueId(appId.get())) {
```

Comment:
Do we need to check the exists? On save, when you get a DuplicateKeyException, you can propogate the GenieConflictException

Reply:
Fair. Saves a query. Will change.